cub/test/catch2_test_device_reduce.cu

```diff
@@ -46,6 +46,21 @@ DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::ArgMin, device_arg_min);
 DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::Max, device_max);
 DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::ArgMax, device_arg_max);
 
+#if defined(__CUDACC_VER_MAJOR__) && defined(__CUDACC_VER_MINOR__) && ((__CUDACC_VER_MAJOR__ > 11) || ((__CUDACC_VER_MAJOR__ == 11) && (__CUDACC_VER_MINOR__ >= 5)))
+#define HAS_NV_PREFIX
+#endif
+// Suppress deprecation warning for the deprecated ArgMin and ArgMax interfaces
+#if defined(HAS_NV_PREFIX)
+_CCCL_NV_DIAG_SUPPRESS(1444)
+#else
+#pragma diag_suppress 1444
+#endif
```

Comment:
Q: Why does `_CCCL_NV_DIAG_SUPPRESS(1444)` not work? It will expand to `#pragma diag suppress 1444` (without nv) when `__NVCC_DIAG_PRAGMA_SUPPORT__` is not defined.